File: backend/app/routes/dolphin.py
from fastapi import APIRouter
from pydantic import BaseModel
from app.services.dolphin_controller import open, close, getGames, fullscreen, pause

router = APIRouter()

class OpenRequest(BaseModel):
    filePath: str

@router.post("/play")
async def openGame(req: OpenRequest):
    return await open(req.filePath)

@router.post("/close")
async def closeDolph():
    result = await close()
    return result

@router.get("/gameList")
def getGameList():
    return getGames()

@router.post("/fullscreen")
def fScreen():
    return fullscreen()

@router.post("/playPause")
def pPause():
    return pause()

File: frontend/src/api/dolphin.ts
import { apiGet, apiPost } from "./client";

type GamesList = {
    [key: string]: string
}

export function play(filePath: string) {
  return apiPost("/dolphin/play", { filePath });
}

export function fullscreen() {
  return apiPost("/dolphin/fullscreen");
}

export function close() {
  return apiPost("/dolphin/close");
}

export function togglePlay() {
    return apiPost("/dolphin/playPause")
}

export function showGamesList(): Promise<GamesList> {
  return apiGet<GamesList>("/dolphin/gameList")
}